server/multitenancy/tenant_resolver.ts

```diff
@@ -98,36 +100,38 @@ function resolve(
   }
 
   if (requestedTenant) {
+    requestedTenant = requestedTenant.toLowerCase();
     if (availableTenants[requestedTenant]) {
       return requestedTenant;
     }
 
     if (
       privateTenantEnabled &&
       availableTenants[username] &&
-      requestedTenant === PRIVATE_TENANT /* || requestedTenant === 'private'*/
+      PRIVATE_TENANTS.indexOf(requestedTenant) > -1
     ) {
-      return PRIVATE_TENANT;
+      return PRIVATE_TENANT_SYMBOL;
     }
 
-    if (
-      globalTenantEnabled &&
-      requestedTenant === GLOBAL_TENANT /* || requestedTenant === 'global'*/
-    ) {
-      return GLOBAL_TENANT;
+    if (globalTenantEnabled && GLOBAL_TENANTS.indexOf(requestedTenant) > -1) {
+      return GLOBAL_TENANT_SYMBOL;
     }
   }
```

Comment:
Just out of curiosity, should the priority of this part be lower than the preferred tenants settings?

Reply:
actually yes, the preferred tenant is a setting in the `kibana.yml`, which is a global setting by administrator. while requested tenant can be tenant from URL parameter or cookie, thus have higher priority than preferred tenant. I know the naming is somewhat confusing. 

preferred tenant setting: https://github.com/opendistro-for-elasticsearch/security-kibana-plugin/blob/master/index.js#L76